sdks/java/harness/src/main/java/org/apache/beam/fn/harness/status/BeamFnStatusClient.java

```diff
@@ -42,9 +44,12 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class BeamFnStatusClient {
+public class BeamFnStatusClient implements AutoCloseable {
+  private static final Object COMPLETED = new Object();
   private final StreamObserver<WorkerStatusResponse> outboundObserver;
   private final BundleProcessorCache processBundleCache;
+  private final ManagedChannel channel;
+  private final CompletableFuture<Object> inboundObserverCompletion;
```

Comment:
Can we use a CompletableFuture<Void> instead?

Reply:
I'm not sure I got what you mean, isn't it already a CompletableFuture.

Reply:
Sorry, I meant a CompletableFuture<Void> since we're not using the returned Object. Not a blocker though

Reply:
Added logging and used the return object.